web/modules/custom/joinup_core/joinup_core.post_update.php

```diff
@@ -551,3 +551,19 @@ function joinup_core_post_update_enable_spdx() {
 function joinup_core_post_update_re_import_legal_type_vocabulary() {
   \Drupal::service('joinup_core.vocabulary_fixtures.helper')->importFixtures('licence-legal-type');
 }
+
+/**
+ * Corrects the versions of faulty news items.
+ */
+function joinup_core_post_update_set_news_default_version() {
+  $results = \Drupal::service('joinup_core.requirements_helper')->getNodesWithProblematicRevisions();
```

Comment:
Can you provide a comment here why this is needed? It is not clear why news items are considered to be "faulty". It is also not dealing with "versions" but with revisions.

This code looks quite dodgy without a proper explanation. This will just take a bunch of news items, update the published revision to the latest one. This is causing data loss and a security risk since if there are news items that have a more recent draft which is not ready for publication, then this code will blindly publish this. Apart from doing a database restore it will be impossible to recover from this action, the information about the currently published revision will be lost forever.

I am not sure what the problem is we are trying to fix but this doesn't seem like a good solution. It is perfectly allowed in core to publish older revisions and have a newer unpublished revision. Why is this needed?